codebase-pipelines/tests/unit.tftest.hcl

```diff
@@ -340,14 +342,109 @@ run "test_codebuild_images_not_required" {
   }
   assert {
     condition     = aws_cloudwatch_event_rule.ecr_image_publish[0].event_pattern == "{\"detail\":{\"action-type\":[\"PUSH\"],\"image-tag\":[\"latest\"],\"repository-name\":[\"my-app/my-codebase\"],\"result\":[\"SUCCESS\"]},\"detail-type\":[\"ECR Image Action\"],\"source\":[\"aws.ecr\"]}"
-    error_message = "Event pattern is incorrect ${jsonencode(aws_cloudwatch_event_rule.ecr_image_publish[0].event_pattern)}"
+    error_message = "Event pattern is incorrect"
   }
   assert {
     condition     = aws_cloudwatch_event_rule.ecr_image_publish[1].event_pattern == "{\"detail\":{\"action-type\":[\"PUSH\"],\"image-tag\":[\"latest\"],\"repository-name\":[\"my-app/my-codebase\"],\"result\":[\"SUCCESS\"]},\"detail-type\":[\"ECR Image Action\"],\"source\":[\"aws.ecr\"]}"
     error_message = "Event pattern is incorrect"
   }
 }
 
+run "test_additional_private_ecr_repository" {
+  command = plan
+
+  variables {
+    additional_ecr_repository = "repository-namespace/repository-name"
+  }
+
+  assert {
+    condition     = local.is_additional_repo_public == false
+    error_message = "Should be: false"
+  }
+  assert {
+    condition     = one(aws_codebuild_project.codebase_image_build[""].environment).environment_variable[4].name == "ADDITIONAL_ECR_REPOSITORY"
+    error_message = "Should be: 'ADDITIONAL_ECR_REPOSITORY'"
+  }
+  assert {
+    condition     = one(aws_codebuild_project.codebase_image_build[""].environment).environment_variable[4].value == "repository-namespace/repository-name"
```

Comment:
I wonder whether it's worth using the approach below that takes away the positional brittleness on these others too?